base/src/main/java/io/spine/validate/Range.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validate;
+
+import io.spine.option.OptionsProto;
+
+/**
+ * A validating option that limits a numeric field to be in the specified range.
+ *
+ * @param <V>
+ *         a value that this option is applied to
+ */
+final class Range<V extends Number & Comparable> extends FieldValidatingOption<String, V> {
+
+    private Range() {
+        super(OptionsProto.range);
+    }
+
+    /** Creates a new instance of this option. */
+    static <V extends Number & Comparable> Range<V> create() {
+        return new Range<>();
+    }
+
+    @Override
+    public Constraint<FieldValue<V>> constraintFor(FieldValue<V> value) {
```

Comment:
Please pass `FieldDescriptor` to methods like this. We don't really need a value.

Reply:
As discussed vocally, let's leave this change outside of the scope of this PR.